InnerEye/ML/dataset/scalar_dataset.py

```diff
@@ -150,10 +158,12 @@ def load_single_data_source(subject_rows: pd.DataFrame,
                             categorical_data_encoder: Optional[CategoricalToOneHotEncoder] = None,
                             metadata_columns: Optional[Set[str]] = None,
                             is_classification_dataset: bool = True,
+                            num_classes: int = 1,
                             sequence_position_numeric: Optional[int] = None) -> T:
     """
     Converts a set of dataset rows for a single subject to a ScalarDataSource instance, which contains the
     labels, the non-image features, and the paths to the image files.
+    :param num_classes: number of classes
```

Comment:
for all places where num_classes is expected, there should be a consistent documentation that says what is expected in binary classification, multi-class, multi-label, regression.